src/message/messagesActions.js

```diff
@@ -34,6 +34,24 @@ export const messageLinkPress = (href: string) => (dispatch: Dispatch, getState:
 
     dispatch(doNarrow(narrow, anchor));
   } else {
-    openLink(getFullUrl(href, auth.realm));
+    let url = getFullUrl(href, auth.realm);
+    if (url.startsWith(auth.realm)) {
+      // get the AWS resource file path and open it in browser
+      const srcPath = url.substring(auth.realm.length);
+      if (
+        srcPath.startsWith('/user_uploads/')
+        || srcPath.startsWith('/thumbnail?')
+        || srcPath.startsWith('/avatar/')
+      ) {
+        const delimiter = url.includes('?') ? '&' : '?';
+        url += `${delimiter}api_key=${auth.apiKey}`;
+
+        fetch(url).then(response => {
+          openLink(response.url);
```

Comment:
This is likely not enough. I don't thick all realms' resources are behind a redirect to URL AWS.